src/app/components/scroller/scroller.ts

```diff
@@ -473,31 +465,38 @@ export class Scroller implements OnInit, AfterContentInit, AfterViewChecked, OnD
     }
 
     ngAfterViewInit() {
-        this.setContentEl(this.contentEl);
-        this.init();
-
-        this.defaultWidth = DomHandler.getWidth(this.elementViewChild.nativeElement);
-        this.defaultHeight = DomHandler.getHeight(this.elementViewChild.nativeElement);
-        this.initialized = true;
+        if (DomHandler.isVisible(this.elementViewChild?.nativeElement)) {
+            this.setInitialState();
+            this.setContentEl(this.contentEl);
+            this.init();
+
+            this.defaultWidth = DomHandler.getWidth(this.elementViewChild.nativeElement);
+            this.defaultHeight = DomHandler.getHeight(this.elementViewChild.nativeElement);
+            this.initialized = true;
+        }
     }
 
     ngAfterViewChecked() {
+        if (!this.initialized) {
+            this.ngAfterViewInit();
```

Comment:
Hi, this looks weird. I think it's better to create a private method and call it instead of using the component's lifecycle method.

Reply:
Hi, thanks a lot for your review! I agree with you. I'll update this line.